### ISSUE TITLE
hostserv: add ability to set valid/invalid vHost patterns for requests

### ISSUE THREAD

**Issue by Renegade334** [zohlai/zohlai#5]
Currently, there is only a rudimentary check on vHost requests through HostServ (`check_vhost_validity`) to check if they are fundamentally valid strings. However, many networks will implement their own vHost policy, such as no capitals, no valid TLDs, or whatever else.

There could be added functionality for customising what HostServ will consider a valid or invalid vHost. It could be implemented similar to NickServ's BADMAIL function, with simple globmask matching of invalid strings. It would probably be better suited to regex matching, which would provide more control over patterns.

**Comment by maxteufel** [comment]
Good idea, but marking as needs-discussion because a network could also write a custom protocol module with a custom check_vhost_validity function.

**Comment by Renegade334** [comment]
A candidate for providing the prerequisite behaviour: Renegade334/atheme@ed4e16c1d9ba289f880d72c1977462e777e71be4. A quick proof-of-concept is at https://gist.github.com/Renegade334/fba75c07fc79eccb1b4c.

**Comment by kaniini** [comment]
Seems kinda ugly and hackish that way, no?

**Comment by Renegade334** [comment]
Well, it's similar in implementation to protocol-related validation functions such as `is_valid_host`. Hackish, yes, but I wouldn't call it ugly! ;-)
